tests/html_whitespace/jsfmt.spec.js

```diff
@@ -1 +1,56 @@
-run_spec(__dirname, ["html"]);
+run_spec(
+  {
+    dirname: __dirname,
+    snippets: [
+      ...[
+        // https://developer.mozilla.org/en-US/docs/Glossary/Whitespace#In_HTML
+        // single
+        "\u0009",
+        "\u000A",
+        "\u000C",
+        "\u000D",
+        "\u0020",
+
+        // many
+        "\u0009\u000A\u000C\u000D\u0020",
+      ].map((textContent) => ({
+        code: `<div>${textContent}</div>`,
+        name: "should be empty",
+      })),
+
+      ...[
+        // single
+        "\u0009",
+        "\u000A",
+        "\u000C",
+        "\u000D",
+        "\u0020",
+
+        // many
+        "\u0009\u000A\u000C\u000D\u0020",
+      ].map((textContent) => ({
+        code: `<span>${textContent}</span>`,
+        name: "should keep one space",
+      })),
+
+      // non-space
+      ...[
+        "\u0005",
+        " \u0005        ",
+        "        \u0005\u0005 ",
+        "        \u0005        \u0005 ",
+      ].map((textContent) => `<div>${textContent}</div>`),
+
+      ...[
+        "\u0005",
+        " \u0005        ",
+        "        \u0005\u0005 ",
+        "        \u0005        \u0005 ",
+      ].map((textContent) => `<span>${textContent}</span>`),
+
+      // #7103
+      "<p><span>X</span> \u0005or\u0005 <span>Y</span></p><p>X \u0005or\u0005 Y</p>",
```

Comment:
This change seems to reproduce #7103 for me:

```suggestion
      "<p><span>X</span> \u0005or\u0005 <span>Y</span></p><p>X \u2005or\u2005 Y</p>",
```

Reply:
Sorry. typo

Reply:
Right, seem the `U+2005` is lost, I'll look into this. Thank you!